### DISCUSSION TITLE
Feature Request - Right-to-Left Reading Support

### DISCUSSION THREAD

**Discussion by BernardByte** [sumatrapdfreader/sumatrapdf#3883] [Ideas]
**Hi Sumatra PDF team,**

I hope you're doing well. I'm a regular user of Sumatra PDF and find it to be an excellent lightweight PDF reader. However, I'd like to propose a feature that could enhance the reading experience for users of languages like Urdu and Arabic.

Feature Request:
I suggest adding a "Right-to-Left Reading" option, similar to Adobe Acrobat's functionality. This would be particularly useful for readers of scripts that are written from right to left.

Implementation:
Introduce an option on right click menu for "Right-to-Left Reading" mode.
Adjust the page rendering to flow from right to left when this mode is activated.

Benefits:
Improved usability for users reading Urdu and Arabic content.
Enhanced inclusivity, making Sumatra PDF a more versatile PDF reader.

Muhammad Imran

**Comment by GitHubRulesOK** [comment]
add a switch to toggle RTL (Manga) mode
I often set mine to M (so measure is lowercase and RTL is uppercase)
It is not just for Japanese nor is it confined to PDF see https://github.com/sumatrapdfreader/sumatrapdf/issues/1264#issuecomment-1150241521

**Reply by BernardByte** [reply]
idk how to add a switch. I'm just a user.

**Comment by kjk** [comment]
It probably already exists as manga mode. Use `Ctrl + K`, `Toggle Manga Mode` command.
You can bind the command to a keyboard shortcut as per https://www.sumatrapdfreader.org/docs/Customizing-keyboard-shortcuts i.e. in advanced settings add:
```
Shortcuts [
	[
		Cmd = CmdToggleMangaMode
		Key = m
	]
]
```

**Reply by GitHubRulesOK** [reply]
m is for measure hence I use Shifted M (Key = M)

**Comment by BernardByte** [comment]
I think problem is resolved so I'm closing this discussion.
